From 13bdc0f9721164b49cb0bf755c151c97cf430d70 Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Wed, 25 Feb 2015 17:49:31 +0100
Subject: [PATCH 1/6] ATN-CPDLC: never put a break in a TRY block

It will skip the execution of ENDTRY block

Bug: 9952
Change-Id: I55af248b7d31cf1b75aae695ef8c149367791820
Reviewed-on: https://code.wireshark.org/review/7391
Petri-Dish: Pascal Quantin <pascal.quantin@gmail.com>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Pascal Quantin <pascal.quantin@gmail.com>
(cherry picked from commit 1a3dd349233a4ee3e69295c8e79f9a216027037e)
Conflicts:
	epan/dissectors/packet-atn-cpdlc.c
Reviewed-on: https://code.wireshark.org/review/7392
Reviewed-on: https://code.wireshark.org/review/7702
Reviewed-by: Peter Wu <peter@lekensteyn.nl>
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
Tested-by: Balint Reczey <balint@balintreczey.hu>
---
 asn1/atn-cpdlc/packet-atn-cpdlc-template.c | 18 ++++++++++--------
 epan/dissectors/packet-atn-cpdlc.c         | 22 ++++++++++++----------
 2 files changed, 22 insertions(+), 18 deletions(-)

diff --git a/asn1/atn-cpdlc/packet-atn-cpdlc-template.c b/asn1/atn-cpdlc/packet-atn-cpdlc-template.c
index b0ba23c..1050754 100644
--- a/asn1/atn-cpdlc/packet-atn-cpdlc-template.c
+++ b/asn1/atn-cpdlc/packet-atn-cpdlc-template.c
@@ -227,17 +227,18 @@ dissect_atn_cpdlc_heur(
 					TRY {
 						dissect_ProtectedGroundPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = TRUE;
-						break;}
+						is_pm = TRUE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
 					ENDTRY;
+					if (is_atn_cpdlc) {
+						break;
+					}
 					TRY {
 		        dissect_GroundPDUs_PDU(tvb, pinfo, NULL);
 						is_pm = FALSE;
-						is_atn_cpdlc = TRUE;
-						break;}
+						is_atn_cpdlc = TRUE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
@@ -247,17 +248,18 @@ dissect_atn_cpdlc_heur(
 					TRY {
 						dissect_ProtectedAircraftPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = TRUE;
-						break;}
+						is_pm = TRUE;}
 					CATCH_ALL {
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE; }
 					ENDTRY;
+					if (is_atn_cpdlc) {
+						break;
+					}
 					TRY{
 						dissect_AircraftPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = FALSE;
-						break;}
+						is_pm = FALSE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
diff --git a/epan/dissectors/packet-atn-cpdlc.c b/epan/dissectors/packet-atn-cpdlc.c
index e21ffa7..526f130 100644
--- a/epan/dissectors/packet-atn-cpdlc.c
+++ b/epan/dissectors/packet-atn-cpdlc.c
@@ -5424,17 +5424,18 @@ dissect_atn_cpdlc_heur(
 					TRY {
 						dissect_ProtectedGroundPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = TRUE;
-						break;}
+						is_pm = TRUE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
 					ENDTRY;
+					if (is_atn_cpdlc) {
+						break;
+					}
 					TRY {
 		        dissect_GroundPDUs_PDU(tvb, pinfo, NULL);
 						is_pm = FALSE;
-						is_atn_cpdlc = TRUE;
-						break;}
+						is_atn_cpdlc = TRUE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
@@ -5444,17 +5445,18 @@ dissect_atn_cpdlc_heur(
 					TRY {
 						dissect_ProtectedAircraftPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = TRUE;
-						break;}
+						is_pm = TRUE;}
 					CATCH_ALL {
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE; }
 					ENDTRY;
+					if (is_atn_cpdlc) {
+						break;
+					}
 					TRY{
 						dissect_AircraftPDUs_PDU(tvb, pinfo, NULL);
 						is_atn_cpdlc = TRUE;
-						is_pm = FALSE;
-						break;}
+						is_pm = FALSE;}
 					CATCH_ALL{
 						is_atn_cpdlc = FALSE;
 						is_pm = FALSE;}
@@ -7918,7 +7920,7 @@ void proto_register_atn_cpdlc (void)
         NULL, HFILL }},
 
 /*--- End of included file: packet-atn-cpdlc-hfarr.c ---*/
-#line 316 "../../asn1/atn-cpdlc/packet-atn-cpdlc-template.c"
+#line 318 "../../asn1/atn-cpdlc/packet-atn-cpdlc-template.c"
 			};
 
 		static gint *ett[] = {
@@ -8069,7 +8071,7 @@ void proto_register_atn_cpdlc (void)
     &ett_atn_cpdlc_WindSpeed,
 
 /*--- End of included file: packet-atn-cpdlc-ettarr.c ---*/
-#line 320 "../../asn1/atn-cpdlc/packet-atn-cpdlc-template.c"
+#line 322 "../../asn1/atn-cpdlc/packet-atn-cpdlc-template.c"
 				&ett_atn_cpdlc
 		};
 
-- 
2.1.4

